src/server/models/cacheable_queries/zip.js

```diff
@@ -0,0 +1,65 @@
+import { getConfig } from "../../api/lib/config";
+import { r } from "..";
+import SmartyStreetsSDK from "smartystreets-javascript-sdk";
+
+// SmartyStreets
+const SmartyStreetsCore = SmartyStreetsSDK.core;
+const Lookup = SmartyStreetsSDK.usZipcode.Lookup;
+
+const clientBuilder = new SmartyStreetsCore.ClientBuilder(
+  new SmartyStreetsCore.StaticCredentials(
+    getConfig("SMARTY_AUTH_ID"),
+    getConfig("SMARTY_AUTH_TOKEN")
+  )
+);
+const client = clientBuilder.buildUsZipcodeClient();
+
+// Cache
+const cacheKey = zip => `${process.env.CACHE_PREFIX || ""}state-of-${zip}`;
+
+const zipStateCache = {
+  clearQuery: async ({ zip }) => {
+    if (r.redis) {
+      await r.redis.delAsync(cacheKey(zip));
+    }
+  },
+  query: async ({ zip }) => {
+    async function getState() {
+      const lookup = new Lookup();
+
+      lookup.zipCode = zip;
+
+      const res = await client.send(lookup);
```

Comment:
just noting we do have a comprehensive static cache of zip to state seeded in the db ([found here](https://raw.githubusercontent.com/MoveOnOrg/Spoke/d43a128a8ec317de024698b88abe343dfd1bd7ec/src/server/seeds/data/zip-codes.csv)), though there may be benefits to using the Smarty API?

Reply:
Oh I didn't even realize this! The benefit of using the Smarty API is up-to-date data. But this is useful to know for future work if we or another org doesn't want to use Smarty API